### PR TITLE
added new query for public lambda

### LINKED ISSUE #1364: Add Public Lambda via API Gateway query for Ansible
### Platform
Ansible

### Provider
AWS

### Description
Allowing to run lambda function using public API Gateway

### PR DESCRIPTION
**Pull request by soniantao**
Closes #1364

### CODE REVIEW

File: assets/queries/ansible/aws/api_gateway_public_lambda/query.rego [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package Cx
+
+CxPolicy [  result ] {
+  document := input.document[i]
+  tasks := getTasks(document)
+  task := tasks[t]
+  awsApiGateway := task["community.aws.lambda"]
+  contains(awsApiGateway.state, "present")
+  role := awsApiGateway.role
+  endswith(role, "/*")
```
Comment (jorge-cruz) [inline assets/queries/ansible/aws/api_gateway_public_lambda/query.rego:10]: I don't think the 'role' of a "community.aws.lambda" task concerns this query.
Instead, I think you should check if a "community.aws.lambda_policy" task gives permission to invoke a lambda function using public API Gateway. Check the implementation of the Terraform's version for this query done by Vasyl (the fields are similar).

**Review by jorge-cruz — CHANGES_REQUESTED** [review]

**Comment by markmishaevcx** [comment]
Scan submitted to Checkmarx

File: assets/queries/ansible/aws/api_gateway_public_lambda/query.rego (hunk 1 above)
Comment (soniantao) [reply assets/queries/ansible/aws/api_gateway_public_lambda/query.rego]: Hi Jorge. I don't know what happen but this query was not related al all indeed. I've changed everything: samples + query.

**Comment by markmishaevcx** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 4 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**3 Medium**<br />Low&nbsp;**1 Low**<br />Info&nbsp;**0 Info**<br />
### Violation Summary
Medium&nbsp;**3** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049018&projectid=707)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[30](https://github.com/Checkmarx/kics/blob/Feature/PublicLambda/cmd/builder/main.go#L30) |Medium|Path_Traversal|cmd/builder/main.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049018&projectid=707&pathid=3)
[52](https://github.com/Checkmarx/kics/blob/Feature/PublicLambda/pkg/kics/service.go#L52) |Medium|Denial_Of_Service_Resource_Exhaustion|pkg/kics/service.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049018&projectid=707&pathid=1)
[83](https://github.com/Checkmarx/kics/blob/Feature/PublicLambda/test/main_test.go#L83) |Medium|Denial_Of_Service_Resource_Exhaustion|test/main_test.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049018&projectid=707&pathid=2)

File: assets/queries/ansible/aws/api_gateway_public_lambda/query.rego [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package Cx
+
+CxPolicy [result] {
+  document := input.document[i]
+  tasks := getTasks(document)
+  task := tasks[t]
+  lambda := task["lambda_policy"]
+  lambda.action == "lambda:InvokeFunction"
```
Comment (jorge-cruz) [inline assets/queries/ansible/aws/api_gateway_public_lambda/query.rego:8]: Check also if `lambda.action == "lambda:*"`

File: assets/queries/ansible/aws/api_gateway_public_lambda/metadata.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "Public_Lambda_via_API_Gateway",
```
Comment (jorge-cruz) [inline assets/queries/ansible/aws/api_gateway_public_lambda/metadata.json:2]: Change the id to lowercase letters: `public_lambda_via_api_gateway`
Also, change the directory name to be equal to the query's id

**Review by jorge-cruz — CHANGES_REQUESTED** [review]

**Comment by markmishaevcx** [comment]
Scan not submitted to Checkmarx due to existing Active scan for the same project.

File: assets/queries/ansible/aws/api_gateway_public_lambda/metadata.json (hunk 3 above)
Comment (jorge-cruz) [reply assets/queries/ansible/aws/api_gateway_public_lambda/metadata.json]: Don't forget to change the directory name to be equal to the query's id

**Review by jorge-cruz — APPROVED** [review]

**Pull request merged**